library/init/meta/has_reflect.lean

```diff
@@ -48,3 +48,9 @@ meta instance prod.reflect {α β : Type} [has_reflect α] [reflected α] [has_r
 
 meta instance pos.reflect : has_reflect pos
 | ⟨l, c⟩ := `(_)
+
+meta def reflected.to_format {α} {a : α} (h : reflected a) : format :=
+to_fmt (reflect a).to_expr
+
+meta instance {α} (a : α) : has_to_format (reflected a) :=
+⟨reflected.to_format⟩
```

Comment:
Logically, these definitions should probably go in `expr.lean` below the definition of `reflected`. That file is already quite big of course, but you can inline the first definition into the second one, which seems to be our standard for (non-recursive) meta instance definitions.